packages/upload-api/src/access/claim.js

```diff
@@ -2,36 +2,144 @@ import * as Server from '@ucanto/server'
 import * as Access from '@web3-storage/capabilities/access'
 import * as API from '../types.js'
 import * as delegationsResponse from '../utils/delegations-response.js'
+import { createSessionProofs } from './confirm.js'
 
 /**
  * @param {API.AccessClaimContext} ctx
  */
 export const provide = (ctx) =>
   Server.provide(Access.claim, (input) => claim(input, ctx))
 
+/**
+ * Checks if the given Principal is an Account.
+ * @param {API.Principal} principal
+ * @returns {principal is API.Principal<API.DID<'mailto'>>}
+ */
+const isAccount = (principal) => principal.did().startsWith('did:mailto:')
+
 /**
  * @param {API.Input<Access.claim>} input
  * @param {API.AccessClaimContext} ctx
  * @returns {Promise<API.Result<API.AccessClaimSuccess, API.AccessClaimFailure>>}
  */
-export const claim = async (
-  { invocation },
-  { delegationsStorage: delegations }
-) => {
+export const claim = async ({ invocation }, { delegationsStorage, signer }) => {
   const claimedAudience = invocation.capabilities[0].with
-  const claimedResult = await delegations.find({ audience: claimedAudience })
-  if (claimedResult.error) {
+  const storedDelegationsResult = await delegationsStorage.find({
+    audience: claimedAudience,
+  })
+
+  if (storedDelegationsResult.error) {
     return {
       error: {
         name: 'AccessClaimFailure',
         message: 'error finding delegations',
-        cause: claimedResult.error,
+        cause: storedDelegationsResult.error,
+      },
+    }
+  }
+
+  // If this Agent has been confirmed by any Accounts, we'll find `*`/`ucan:*`
+  // delegations for them. But they won't have any proofs on them. They're proof
+  // of login, but don't serve to give the Agent any actual capabilities. That's
+  // our job.
+  const loginDelegations = storedDelegationsResult.ok.filter((delegation) => {
```

Comment:
So, it's my understanding that `access/delegate` and `access/claim` are general purpose. I think it's ok to add functionality from `access/confirm` here but it would be nice (slash not a breaking change) to retain existing ability to stash/retrieve any delegation. I think with the current implementation you'd not receive anything unless `ucan:*` delegations are found?

Reply:
Ah, yes, that's true. We should just translate those and leave the rest in the output.